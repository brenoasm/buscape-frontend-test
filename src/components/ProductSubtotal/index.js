import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { colors } from '../../theme';

import { numberToReal } from '../../utils/converters';

const propTypes = {
  price: PropTypes.object
};

const defaultProps = {
  price: {
    installments: 0,
    installmentValue: 0,
    value: 0
  }
};

const StyledProductSubtotal = styled.div`
  > span:first-child {
    font-size: 22px;
    font-weight: 600;
    line-height: 35px;
    color: ${colors.green};

    > span {
      font-size: 32px;
    }
  }

  > span:last-child {
    > span {
      color: ${colors.green};
      font-weight: 600;
    }
  }
`;

const ProductSubtotal = ({
  price
}) => (
  <StyledProductSubtotal>
    <span>
      {price.installments}x R${" "}
      <span>{numberToReal(price.installmentValue)}</span>
    </span>
    <span>
      <br />
      ou <span>R$ {numberToReal(price.value)}</span> Ã  vista
    </span>
  </StyledProductSubtotal>
);

ProductSubtotal.propTypes = propTypes;
ProductSubtotal.defaultProps = defaultProps;

export default ProductSubtotal;
