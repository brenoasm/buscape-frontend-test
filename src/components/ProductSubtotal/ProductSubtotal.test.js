import React from "react";
import { shallow } from "enzyme";

import ProductSubtotal from "./";

describe("ProductSubtotal", () => {
  const price = {
    installments: 5,
    installmentValue: 500,
    value: 2500
  }

  it("matches the snapshot", () => {
    const wrapper = shallow(<ProductSubtotal price={price} />);

    expect(wrapper).toMatchSnapshot();
  });
});
