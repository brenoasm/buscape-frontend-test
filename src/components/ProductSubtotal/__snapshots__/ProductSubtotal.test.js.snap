// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSubtotal matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductSubtotal
    price={
      Object {
        "installmentValue": 500,
        "installments": 5,
        "value": 2500,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <span>
          5
          x R$
           
          <span>
            500,00
          </span>
        </span>,
        <span>
          <br />
          ou 
          <span>
            R$ 
            2.500,00
          </span>
           à vista
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            5,
            "x R$",
            " ",
            <span>
              500,00
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          5,
          "x R$",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "500,00",
            },
            "ref": null,
            "rendered": "500,00",
            "type": "span",
          },
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <br />,
            "ou ",
            <span>
              R$ 
              2.500,00
            </span>,
            " à vista",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "ou ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "R$ ",
                "2.500,00",
              ],
            },
            "ref": null,
            "rendered": Array [
              "R$ ",
              "2.500,00",
            ],
            "type": "span",
          },
          " à vista",
        ],
        "type": "span",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  > span:first-child {
    font-size: 22px;
    font-weight: 600;
    line-height: 35px;
    color: ",
          "#22B14C",
          ";

    > span {
      font-size: 32px;
    }
  }

  > span:last-child {
    > span {
      color: ",
          "#22B14C",
          ";
      font-weight: 600;
    }
  }
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <span>
            5
            x R$
             
            <span>
              500,00
            </span>
          </span>,
          <span>
            <br />
            ou 
            <span>
              R$ 
              2.500,00
            </span>
             à vista
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              5,
              "x R$",
              " ",
              <span>
                500,00
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            5,
            "x R$",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "500,00",
              },
              "ref": null,
              "rendered": "500,00",
              "type": "span",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <br />,
              "ou ",
              <span>
                R$ 
                2.500,00
              </span>,
              " à vista",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "ou ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "R$ ",
                  "2.500,00",
                ],
              },
              "ref": null,
              "rendered": Array [
                "R$ ",
                "2.500,00",
              ],
              "type": "span",
            },
            " à vista",
          ],
          "type": "span",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  > span:first-child {
    font-size: 22px;
    font-weight: 600;
    line-height: 35px;
    color: ",
            "#22B14C",
            ";

    > span {
      font-size: 32px;
    }
  }

  > span:last-child {
    > span {
      color: ",
            "#22B14C",
            ";
      font-weight: 600;
    }
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
